include/mega/command.h

```diff
@@ -639,12 +639,14 @@ class MEGA_API CommandSetShare : public Command
     string msg;
     string personal_representation;
 
+    std::function<void(Error)> completion;
```

Comment:
Maybe we can name it `callback`? (or `mCallback`, indeed)

Reply:
And also, should we simply add this new member to `Command` class? I guess the cost of keeping this object for every command shouldn't be a problem. And it will be available for any new command, just by design. Perhaps a pointer, so we avoid the instantiation :thinking:

Reply:
I don't think we should add it to the command class, because different commands will have different parameters to pass to the completion function - some may pass a string result along with an Error for example, or node handles etc.   Not all will only need an Error, though that is the most common case.   Since the function signature is part of the type, we can't support any variation if we put it in the base class.